packages/core-data/src/actions.js

```diff
@@ -799,7 +799,10 @@ export const saveEditedEntityRecord =
 			recordId
 		);
 		const record = { [ entityIdKey ]: recordId, ...edits };
-		return await dispatch.saveEntityRecord( kind, name, record, options );
+		return await {
+			values: dispatch.saveEntityRecord( kind, name, record, options ),
+			metaChange: !! edits?.meta,
+		};
```

Comment:
I don't think we should touch core-data for this change. I think core-data shouldn't have flags like that that are specific to given entities.